cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -156,13 +191,36 @@ func (ns *nodeSyncer) process(key string) error {
 	if err != nil {
 		return fmt.Errorf("failed to get Pod %q: %v", key, err)
 	}
-	if !exists {
-		node, gsa, found := ns.nodes.remove(key)
-		if !found {
+	if !exists { // pod removal event
+		podUIDs := ns.nodes.find(key)
+		if len(podUIDs) == 0 {
+			klog.Warningf("Pod key %q not found", key)
+			return nil // no retry
+		}
+		if ns.delayGSARemove {
+			for _, uid := range podUIDs {
```

Comment:
In my testing, I found 
1. podUIDs here can have duplicates. 
2. A new removal will go over all existing removals. So if in 30 minutes, we have N pods with the same kns/ksa, we will  see at least N * (N+1)/2 lines of logs "Pod %q (UID %q) is queued for delayed removal".
3. It is fine to add a pod multiple times here since "if an item is added multiple times before it can be processed, it
    will only be processed once." (https://pkg.go.dev/k8s.io/client-go/util/workqueue)

I created/deleted dppod0 3 times and below are the related logs.

I0921 06:03:40.402099       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "4ed10df1-6336-4a18-865b-87c897b66dd6") is queued for delayed removal
I0921 06:03:40.402177       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "4ed10df1-6336-4a18-865b-87c897b66dd6") is queued for delayed removal


I0921 06:07:43.693520       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "4ed10df1-6336-4a18-865b-87c897b66dd6") is queued for delayed removal
I0921 06:07:43.693607       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "4ed10df1-6336-4a18-865b-87c897b66dd6") is queued for delayed removal
I0921 06:07:43.693618       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "677529d4-bbdc-44e9-a822-01e5f4345777") is queued for delayed removal

I0921 06:11:35.456093       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "4ed10df1-6336-4a18-865b-87c897b66dd6") is queued for delayed removal
I0921 06:11:35.456836       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "677529d4-bbdc-44e9-a822-01e5f4345777") is queued for delayed removal
I0921 06:11:35.456866       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "37d8d605-b7e5-49e2-a626-225f30f9af22") is queued for delayed removal
I0921 06:11:35.456877       1 node_syncer.go:202] Pod "kns0/dppod0" (UID "4ed10df1-6336-4a18-865b-87c897b66dd6") is queued for delayed removal

I0921 06:33:40.415811       1 node_syncer.go:170] Pod "4ed10df1-6336-4a18-865b-87c897b66dd6" running as "testgsa0@wanxiny-gke-dev.iam.gserviceaccount.com" is removed from Node "" after
delay

I0921 06:37:43.694709       1 node_syncer.go:170] Pod "677529d4-bbdc-44e9-a822-01e5f4345777" running as "testgsa0@wanxiny-gke-dev.iam.gserviceaccount.com" is removed from Node "gke-dp-t
est-default-pool-7f0a95e1-84j7" after delay

I0921 06:41:35.457907       1 node_syncer.go:170] Pod "37d8d605-b7e5-49e2-a626-225f30f9af22" running as "testgsa0@wanxiny-gke-dev.iam.gserviceaccount.com" is removed from Node "gke-dp-t
est-default-pool-7f0a95e1-84j7" after delay

Reply:
Discussed with Wanxin again.  The impact seems to be excessive log message for now.  We agreed to let this go in for now and look into explaining the additional "queued for delayed removal" once we get more data from different usage scenario.